lib/providers/users.dart

```diff
@@ -48,19 +56,43 @@ class Users with ChangeNotifier {
     return id;
   }
 
-  User getCurrentUser() {
-    var users = _users.values;
-    for (var user in users) {
-      if (user.isCurrentUser) {
-        return user;
-      }
-    }
-    return null;
+  User get currentUser {
+    return _currentUser;
   }
 
-  void deleteUser (int id) {
-    delete(table: 'users', id:id);
+
+  void deleteUser(int id) {
+    delete(table: 'users', id: id);
     _users.remove(id);
     notifyListeners();
   }
+
+  void updateLocation({int id, double latitude, double longitude}) {
+    User user = _users.update(
+      id,
+      (oldUser) => User(
+        id: oldUser.id,
+        name: oldUser.name,
+        isCurrentUser: oldUser.isCurrentUser,
+        latitude: latitude.toString(),
+        longitude: longitude.toString(),
+      ),
+    );
+    update(table: 'users', object: user);
+    notifyListeners();
+  }
+
+  List<Map<String, dynamic>> getUsersLocations() {
+    List<Map<String, dynamic>> userLocations = [];
+    for (var user in _users.values) {
```

Comment:
Do we need to iterate through every user every time? What if I only want to track one person? If there are 50+ people using the system (some concert staff or something) then this could be slow/ the compass may get crowded.

Reply:
I'm going to add a select for the users the main user would like to keep track of, right now it just displays everyone. I'll add it to trello